app/controllers/messages_controller.rb

```diff
@@ -1,5 +1,32 @@
 class MessagesController < ApplicationController
+  before_action :authenticate_admin!, only: [:index]
+
+  MESSAGES_PER_PAGE = 4
+  def show
+  end
+
+  def new
+    @message = Message.new
+  end
+
+  def create
+    @message = Message.new params.require(:message).permit(:email, :content)
+
+    if @message.save
+      redirect_to message_path(@message)
+    else
+      render :new
+    end
+  end
+
   def index
-    @messages = Message.order(created_at: :desc)
+    @messages = Message.page(1).per(MESSAGES_PER_PAGE)
+  end
+
+  def destroy
+    m=Message.find params[:id]
```

Comment:
`message` instead of `m`. Always be explicit.

Also, spaces on either side of `=`. `message = Message...`